### DISCUSSION TITLE
ERROR: Failed to create session.

### DISCUSSION THREAD

**Discussion by astinbiju44** [appium/appium#19995] [Q&A]
`Failed to create session. A new session could not be created. Details: Unable to build Espresso server - Process ended with exitcode 1 (cmd: '/tmp/espresso-server-emulator-5554/gradlew -PappiumTargetPackage\=com.android.settings app\:assembleAndroidTest') Gradle error message: [STDERR] * What went wrong:, [STDERR] Execution failed for task ':app:dexBuilderDebugAndroidTest'., [STDERR] > There was a failure while executing work items, [STDERR] > A failure occurred while executing com.android.build.gradle.internal.dexing.DexWorkAction, [STDERR] > Failed to process: /tmp/espresso-server-emulator-5554/app/build/tmp/kotlin-classes/debugAndroidTest, /tmp/espresso-server-emulator-5554/app/build/intermediates/javac/debugAndroidTest/classes, [STDERR] , [STDERR] * Try:, [STDERR] > Run with --stacktrace option to get the stack trace., [STDERR] > Run with --info or --debug option to get more log output., [STDERR] > Run with --scan to get full insights., [STDERR] , [STDERR] * Get more help at https://help.gradle.org, [STDERR] , [STDERR] BUILD FAILED in 10s, [STDOUT] 32 actionable tasks: 10 executed, 22 from cache. Check https://github.com/appium/appium-espresso-driver#troubleshooting regarding advanced session startup troubleshooting.`

I tried adding espresso, and some error popped up. Anyone got any idea please help?

**Comment by mykola-mokhnach** [comment]
> cmd: '/tmp/espresso-server-emulator-5554/gradlew -PappiumTargetPackage\=com.android.settings app\:assembleAndroidTest'

Espresso driver cannot automate built-in apps. It must instrument an .apk first to be able to interact with it. Check https://github.com/appium/appium-espresso-driver?tab=readme-ov-file#requirements for more details
